### DISCUSSION TITLE
Apple mps backend support

### DISCUSSION THREAD

**Discussion by callankuo** [stanfordnlp/dspy#1153] [Q&A]
I use Mac M3 chip and test the Predict module got RuntimeError: Placeholder storage has not been allocated on MPS device!

**Comment by callankuo** [comment]
Is this means that we can't use Mac book pro for DSPY development? ( I use meta-llama/Llama-2-13b-chat-hf model already cache on my local machine.)

**Comment by setren** [comment]
Got the same issue, any updates 🤔

**Reply by callankuo** [reply]
Use Ollama framework will works.

import dspy
import torch
# Apple device MPS support
device = torch.device("mps")
llama = dspy.OllamaLocal("llama3:8b", max_tokens=5000)
colbertv2_wiki17_abstracts = dspy.ColBERTv2(url='http://20.102.90.50:2017/wiki17_abstracts')
dspy.settings.configure(lm=llama, rm=colbertv2_wiki17_abstracts, device_map = device, trace=[], temperature=0.7)

**Reply by setren** [reply]
It works, cool 🙌
